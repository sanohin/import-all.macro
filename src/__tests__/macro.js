import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'
import prettier from 'prettier'
import {prettier as prettierConfig} from 'kcd-scripts/config'

const projectRoot = path.join(__dirname, '../../').replace(/\\/g, '/')

expect.addSnapshotSerializer({
  print(val) {
    return val
      .split(projectRoot)
      .join('<PROJECT_ROOT>/')
      .replace(/fixtures/g, 'files')
      .replace(/..\/macro/, 'named-import-all.macro')
  },
  test(val) {
    return typeof val === 'string'
  },
})

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(
      result,
      Object.assign({trailingComma: 'es5'}, prettierConfig),
    )
  },
  tests: {
    'no usage': `import importAll from '../macro'`,
    'incorrect API usage': {
      error: true,
      code: `
        import importAll from '../macro'
        const x = importAll.defered('hi')
      `,
    },
    'non-static evaluate-able expression': {
      error: true,
      code: `
        import importAll from '../macro'
        const x = importAll(global.whatever)
      `,
    },
    'README:1 `importAll.deferredNamed` gives an object where key is the named export and value is the dynamic import': `
    import importAll from '../macro'

    const routes = importAll.deferredNamed('./fixtures/*.js')
    `,
  },
})
