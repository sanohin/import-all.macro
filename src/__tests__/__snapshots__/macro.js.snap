// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros README:1 \`importAll.deferredNamed\` gives an object where key is the named export and value is the dynamic import: README:1 \`importAll.deferredNamed\` gives an object where key is the named export and value is the dynamic import 1`] = `

import importAll from 'named-import-all.macro'

const routes = importAll.deferredNamed('./files/*.js')

      ↓ ↓ ↓ ↓ ↓ ↓

const routes = {
  foo: function() {
    return import('./files/a.js').then(function(res) {
      return res.foo
    })
  },
  bar: function() {
    return import('./files/b.js').then(function(res) {
      return res.bar
    })
  },
  Baz: function() {
    return import('./files/c.js').then(function(res) {
      return res.Baz
    })
  },
  A: function() {
    return import('./files/d.js').then(function(res) {
      return res.A
    })
  },
}


`;

exports[`macros incorrect API usage: incorrect API usage 1`] = `

import importAll from 'named-import-all.macro'
const x = importAll.defered('hi')

      ↓ ↓ ↓ ↓ ↓ ↓

Error: <PROJECT_ROOT>/src/__tests__/macro.js: ../macro: This is not supported: \`importAll.defered\`. Please see the named-import-all.macro documentation

`;

exports[`macros no usage: no usage 1`] = `

import importAll from 'named-import-all.macro'

      ↓ ↓ ↓ ↓ ↓ ↓



`;

exports[`macros non-static evaluate-able expression: non-static evaluate-able expression 1`] = `

import importAll from 'named-import-all.macro'
const x = importAll(global.whatever)

      ↓ ↓ ↓ ↓ ↓ ↓

Error: <PROJECT_ROOT>/src/__tests__/macro.js: ../macro: There was a problem evaluating the value of the argument for the code: x = importAll(global.whatever). If the value is dynamic, please make sure that its value is statically deterministic.

`;
